app/src/main/java/org/oppia/app/player/state/StateFragmentPresenter.kt

```diff
@@ -121,7 +167,38 @@ class StateFragmentPresenter @Inject constructor(
 
   private fun subscribeToCurrentState() {
     ephemeralStateLiveData.observe(fragment, Observer<EphemeralState> { result ->
-      logger.d("StateFragment", "getCurrentState: ${result.state.name}")
+      if (result.hasState()) {
+        itemList.clear()
+        currentEphemeralState.set(result)
+
+        updateDummyStateName()
+
+        val interactionId = result.state.interaction.id
+        val hasPreviousState = result.hasPreviousState
+        var hasNextState = false
+        hasGeneralContinueButton = false
+
+        if (!result.terminalState) {
+          if (result.stateTypeCase.number == EphemeralState.COMPLETED_STATE_FIELD_NUMBER
+            && !oldStateNameList.contains(currentEphemeralState.get()!!.state.name)
+          ) {
+            hasGeneralContinueButton = true
+            hasNextState = false
```

Comment:
I'm not sure I follow this--shouldn't there always be a next state if the current state isn't terminal? Or is this also representing the case when you submit the wrong answer? If so, maybe ``canContinueToNextState`` would be a clearer name for this variable.

Reply:
Updated: 
`hasNextState` to `canContinueToNextState`